### PR TITLE
Added class to obtain packages difference

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #440 
Added class to obtain difference between packages present in repo and packages, listed in primary xml.
Also, corrected `XmlPrimaryChecksums` to read `location` tag instead of name, this tag stores real name (file name or storage item name of the package) instead of project name.

### CODE REVIEW

File: src/main/java/com/artipie/rpm/misc/PackagesDiff.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+/**
+ * Differences of packages, listed in primary and packages in the repository.
+ * @since 1.10
+ */
+public final class PackagesDiff {
+
+    /**
+     * Packages, written in primary xml.
+     * Packages file name <-> checksum map.
```
Comment (g4s8) [inline src/main/java/com/artipie/rpm/misc/PackagesDiff.java:20]: `<` and `>` symbols looks wired in Javadoc browser (it uses simplified HTML), so it should be `&lt;-&gt;` to display `<->`

File: src/main/java/com/artipie/rpm/misc/PackagesDiff.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    private final Map<String, String> primary;
+
+    /**
+     * Packages, located in the repository.
+     * Packages file name <-> checksum map.
```
Comment (g4s8) [inline src/main/java/com/artipie/rpm/misc/PackagesDiff.java:26]: Same here.

**Review by g4s8 — CHANGES_REQUESTED** [review]
@olenagerasimova looks good, just a minor comment about Javadoc formatting

**Comment by olenagerasimova** [comment]
@g4s8 thanks

**Review by g4s8 — APPROVED** [review]
@olenagerasimova thanks

**Pull request merged**